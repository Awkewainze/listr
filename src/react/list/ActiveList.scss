@import '../../variables';
@import '../../mixins';

.ActiveList {
  .ListItem {
    transition: opacity 200ms ease;

    @include transitionState($exiting) {
      opacity: 0;
    }
  }
}