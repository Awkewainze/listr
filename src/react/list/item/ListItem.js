import React from 'react';
import './ListItem.scss';
import Checkbox from './Checkbox';
import DeleteButton from './DeleteButton';

const ListItem = ({
  item,
  itemId,
  index,
  checked = false,
  onCheck,
  onDelete
}) => (
  <div className="ListItem">
    <Checkbox className="ListItem__checkbox" checked={checked} index={index} itemId={itemId} onChange={onCheck} />
    <DeleteButton itemId={itemId} onDelete={onDelete} />
    <p className={`ListItem__text ${checked ? 'ListItem__text--checked' : ''}`}>
      {item}
    </p>
  </div>
);

export default ListItem;