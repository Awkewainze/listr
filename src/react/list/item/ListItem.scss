@import '../../../variables';
@import '../../../mixins';

.ListItem {
  display: flex;
  border-bottom: 1px solid #afafaf;
  margin: 6px 0;
  justify-content: flex-start;
  padding: 0 20px;

  @include minWidth($size-mobile) {
    .DeleteButton {
      opacity: 0;
    }

    &:hover {
      border-color: #7f7f7f;

      .DeleteButton {
        opacity: 1;
      }
    }
  }

  .ListItem__text {
    padding: 20px 0;
  }

  .ListItem__checkbox {
    cursor: pointer;
    margin-right: 12px;
  }

  .ListItem__text {
    flex: 1;
    overflow: scroll;
    margin: 12px 0;
    line-height: 26px;
  }

  .ListItem__text--checked {
    text-decoration: line-through;
    opacity: 0.75;
  }

  @include minWidth($size-mobile) {
    .ListItem__checkbox {
      min-width: 72px;
      margin-right: 0;
    }
  }
}