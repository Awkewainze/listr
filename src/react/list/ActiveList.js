import React from 'react';
import './ActiveList.scss';
import ListItem from './item/ListItem';
import TransitionGroup from 'react-transition-group/cjs/TransitionGroup';
import AppCSSTransition from '../common/AppCSSTransition';

const ActiveList = ({ items, onDelete, onToggle }) => (
  <TransitionGroup className="ActiveList">
    {items.map((item, i) =>
      <AppCSSTransition key={item.id} prefix="ListItem" timeout={200}>
        <ListItem
          key={item.id}
          item={item.text}
          checked={item.checked}
          onDelete={onDelete}
          index={i}
          itemId={item.id}
          onCheck={onToggle}
        />
      </AppCSSTransition>
    )}
  </TransitionGroup>
);

export default ActiveList;