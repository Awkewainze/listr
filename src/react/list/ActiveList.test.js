import React from 'react';
import { shallow } from 'enzyme';
import ActiveList from './ActiveList';
import ListItem from './item/ListItem';

const testItems = [
  {
    text: 'apples',
    checked: false,
    id: '123123'
  },
  {
    text: 'oranges',
    checked: true,
    id: '321321'
  }
];

const TEST_PROPS = {
  items: testItems,
  onToggle: jest.fn(),
  onDelete: jest.fn()
};

it('matches snapshot', () => {
  expect(shallow(<ActiveList {...TEST_PROPS} />)).toMatchSnapshot();
});

it('renders items correctly', () => {
  const container = shallow(<ActiveList {...TEST_PROPS} />);
  container.find(ListItem).forEach((listItem, i) => {
    expect(i).toBeLessThan(testItems.length);
    const { checked, text } = testItems[i];
    expect(listItem.prop('checked')).toEqual(checked);
    expect(listItem.prop('item')).toEqual(text);
    expect(listItem.prop('onCheck')).toEqual(TEST_PROPS.onToggle);
    expect(listItem.prop('onDelete')).toEqual(TEST_PROPS.onDelete);
  });
});