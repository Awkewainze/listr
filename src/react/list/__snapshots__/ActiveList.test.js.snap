// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<TransitionGroup
  childFactory={[Function]}
  className="ActiveList"
  component="div"
>
  <AppCSSTransition
    key="123123"
    prefix="ListItem"
    timeout={200}
  >
    <ListItem
      checked={false}
      index={0}
      item="apples"
      itemId="123123"
      key="123123"
      onCheck={[MockFunction]}
      onDelete={[MockFunction]}
    />
  </AppCSSTransition>
  <AppCSSTransition
    key="321321"
    prefix="ListItem"
    timeout={200}
  >
    <ListItem
      checked={true}
      index={1}
      item="oranges"
      itemId="321321"
      key="321321"
      onCheck={[MockFunction]}
      onDelete={[MockFunction]}
    />
  </AppCSSTransition>
</TransitionGroup>
`;
